bridge.js

```diff
@@ -1,151 +1,150 @@
-		case 'exitAck':
-			phantom.exit();
-			break;
-		default:
-			console.error('unrecognized request:'+request);
-			break;
-		}
-	}
-	else{
-		var id=request[0];
-		var page=pages[id];
-		switch(request[2]){
-		case 'pageOpen':
-			page.open(request[3]);
-			break;
-		case 'pageOpenWithCallback':
-			var timeoutId = window.setTimeout(function() {
-				respond([id, cmdId, 'pageOpened', 'timeout']);
-			}, page.settings.resourceTimeout);
-			page.open(request[3], function(status){
-				clearTimeout(timeoutId);
-				respond([id, cmdId, 'pageOpened', status]);
-			});
-			break;
-		case 'pageClose':
-			page.close();
-			respond([id,cmdId,'pageClosed']);
-			break;
-		case 'pageInjectJs':
-			var result=page.injectJs(request[3]);
-			respond([id,cmdId,'pageJsInjected',JSON.stringify(result)]);
-			break;
-		case 'pageIncludeJs':
-			var alreadyGotCallback=false;
-			page.includeJs(request[3], function(){
-				if(alreadyGotCallback)return;
-				respond([id,cmdId,'pageJsIncluded']);
-				alreadyGotCallback=true;
-			});
-			break;
-		case 'pageSendEvent':
-			page.sendEvent(request[3],request[4],request[5]);
-			respond([id,cmdId,'pageEventSent']);
-			break;
-		case 'pageUploadFile':
-			page.uploadFile(request[3],request[4]);
-			respond([id,cmdId,'pageFileUploaded']);
-			break;
-		case 'pageEvaluate':
-			var result=page.evaluate.apply(page,request.slice(3));
-			respond([id,cmdId,'pageEvaluated',JSON.stringify(result)]);
-			break;
-        case 'pageEvaluateAsync':
-			page.evaluateAsync.apply(page,request.slice(3));
-			respond([id,cmdId,'pageEvaluatedAsync']);
-			break;
-		case 'pageRender':
-			page.render(request[3]);
-			respond([id,cmdId,'pageRendered']);
-			break;
-		case 'pageRenderBase64':
-			var result=page.renderBase64(request[3]);
-			respond([id,cmdId,'pageRenderBase64Done', result]);
-			break;
-		case 'pageSet':
-			page[request[3]]=request[4];
-			respond([id,cmdId,'pageSetDone']);
-			break;
-		case 'pageGet':
-			var result=page[request[3]];
-			respond([id,cmdId,'pageGetDone',JSON.stringify(result)]);
-			break;
-		case 'pageSetFn':
-			page[request[3]] = eval('(' + request[4] + ')')
-			break;
-		case 'pageSetViewport':
-			page.viewportSize = {width:request[3], height:request[4]};
-			respond([id,cmdId,'pageSetViewportDone']);
-			break;
-		default:
-			console.error('unrecognized request:'+request);
-			break;
-		}
-	}
-	//console.log('command:'+parts[1]);
-	return;
-};
+controlpage.onAlert = function (msg) {
+  var request = JSON.parse(msg)
+  var cmdId = request[1]
+  var result = null
+  var id = null
+  var page = null
+  // console.log('Request: ' + request);
+  if (request[0] === 0) {
+    switch (request[2]) {
+      case 'createPage':
+        id = pageId++
+        page = webpage.create()
+        pages[id] = page
+        setupPushNotifications(id, page)
+        respond([id, cmdId, 'pageCreated'])
+        break
+      case 'injectJs':
+        var success = phantom.injectJs(request[3]) // eslint-disable-line
```

Comment:
Ok great will do. I will investigate adding the config file for the next commit.